### PR TITLE
feat: add currencies pallet

### PR DESCRIPTION
**Pull request by Roznovjak**
Version from polkadot-v0.9.19 (the latest version with `Events`) with dependencies upgraded to 0.9.26

### CODE REVIEW

**Comment by lumir-mrkva** [comment]
Task linked: [CU-2pjr0p0 Currencies pallet with events](https://app.clickup.com/t/2pjr0p0)

File: currencies/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,734 @@
...
+        pub fn transfer_native_currency(
+            origin: OriginFor<T>,
+            dest: <T::Lookup as StaticLookup>::Source,
+            #[pallet::compact] amount: BalanceOf<T>,
+        ) -> DispatchResult {
+            let from = ensure_signed(origin)?;
+            let to = T::Lookup::lookup(dest)?;
+            T::NativeCurrency::transfer(&from, &to, amount)?;
+
+            Self::deposit_event(Event::Transferred {
```
Comment (dmoka) [inline currencies/src/lib.rs:181]: I would extend one of the corresponding tests with asserting that this event is emitted.

File: currencies/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,734 @@
...
+impl<T: Config> MultiCurrencyExtended<T::AccountId> for Pallet<T> {
+    type Amount = AmountOf<T>;
+
+    fn update_balance(currency_id: Self::CurrencyId, who: &T::AccountId, by_amount: Self::Amount) -> DispatchResult {
+        if currency_id == T::GetNativeCurrencyId::get() {
+            T::NativeCurrency::update_balance(who, by_amount)?;
+        } else {
+            T::MultiCurrency::update_balance(currency_id, who, by_amount)?;
+        }
+        Self::deposit_event(Event::BalanceUpdated {
```
Comment (dmoka) [inline currencies/src/lib.rs:335]: I would also extend a corresponding test with asserting this deposited event

File: currencies/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,734 @@
...
+        }
+    }
+
+    fn transfer(
+        currency_id: Self::CurrencyId,
+        from: &T::AccountId,
+        to: &T::AccountId,
+        amount: Self::Balance,
+    ) -> DispatchResult {
+        if amount.is_zero() || from == to {
```
Comment (dmoka) [inline currencies/src/lib.rs:258]: I would add an edge test case for transferring this `from == to`

File: currencies/src/tests.rs [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+
+#![cfg(test)]
+
+use super::*;
+use frame_support::{assert_noop, assert_ok};
+use mock::{Event, *};
+use sp_runtime::traits::BadOrigin;
+
+#[test]
+fn multi_lockable_currency_should_work() {
```
[the remark this answers is not in the document]
Comment (Roznovjak) [reply currencies/src/tests.rs:11]: I would like to verify with @lumir-mrkva that it's ok to modify the original implementation.

**Pull request merged**